core/src/main/scala/org/ensime/indexer/domain.scala

```diff
@@ -47,27 +47,34 @@ case class ClassName(pack: PackageName, name: String) extends FullyQualifiedName
     if (pack.path.isEmpty) name
     else ClassName.cleanupPackage(pack.fqnString + "." + name)
 
-  def internalString =
-    "L" + (if (pack.path.isEmpty) name else pack.path.mkString("/") + "/" + name) + ";"
+  def internalString = name match {
```

Comment:
perhaps cleaner, you could introduce a new parameter `internalString` into `ClassName` and then provide an `apply` constructor for the `ClassName` which will construct the longer form `L...;` name.